deepwell/src/methods/user.rs

```diff
@@ -36,7 +39,8 @@ pub async fn user_get(req: ApiRequest) -> ApiResponse {
     let reference = ItemReference::try_from(&req)?;
     let user = req.user(&txn).get(reference).await.to_api()?;
     txn.commit().await?;
-    build_user_response(&user, StatusCode::Ok)
+    let query: UserDetailsQuery = req.query().unwrap_or_default();
```

Comment:
Since `req.query()` returns `Result<_>`, wouldn't this mean that a malformed details param yields implicit default instead of an error?

Annotate `detail` in `UserDetailsQuery` with `#[serde(default)]` and then you can just do `req.query()?` and handle errors but get defaults as expected.

This is very minor, but I'm not super big on `Default` being implemented for `UserDetailsQuery`.

Reply:
Optional suggestion:
```suggestion
    let UserDetailsQuery { details } = req.query().unwrap_or_default();
```

Reply:
Despite this being internal, I am following the external OpenAPI spec. Kind of. I mean, should a malformed query always return an error? I guess in an API context it should, but usually with query parameters you just ignore them if they don't make any sense.

Reply:
Ignoring unknown query parameters are fine but idk about ignoring an invalid value for a real key.

Reply:
Also, with the `?` operator, what happens with an empty query string? Does `UserDetailsQuery.details` have to be `Option<UserDetails>` or will that implicitly have `default()` called on it?

Reply:
No if it's empty it gets filled in because you add `#[serde(default)]` to its field in the struct definition.

It says "when trying to deserialize, if this field is missing then just fill it with the default", which is what we want.

As for `?`, it is the same as the `try!()` macro, which is the same as: (effectively)

```rs
let item = try!(result);

// same as

let item = match result {
    Ok(item) => item,
    Err(error) => return Err(error.into()),
};
```

It's syntactic sugar for dealing with operations that can fail.

Reply:
Okay, I'll do what you're suggesting then.